Services/SystemServer/main.cpp

```diff
@@ -84,6 +85,86 @@ static void parse_boot_mode()
     dbg() << "Booting in " << g_boot_mode << " mode";
 }
 
+static void prepare_devfs()
+{
+    // FIXME: Find a better way to all of this stuff, without hardcoding all of this!
+
+    int rc = mount(-1, "/dev", "dev", 0);
+    if (rc != 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = mkdir("/dev/pts", 0755);
+    if (rc != 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = mount(-1, "/dev/pts", "devpts", 0);
+    if (rc != 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = symlink("/proc/self/fd/0", "/dev/stdin");
+    if (rc < 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = symlink("/proc/self/fd/1", "/dev/stdout");
+    if (rc < 0) {
+        ASSERT_NOT_REACHED();
+    }
+    rc = symlink("/proc/self/fd/2", "/dev/stderr");
+    if (rc < 0) {
+        ASSERT_NOT_REACHED();
+    }
```

Comment:
These are not necessary. ProcFS will generate them :)

Reply:
This creates `/dev/stdin -> /proc/self/fd/0`, not the other way around 🙂

Reply:
oh I see, so `build-root-filesystem.sh` created these links for nothing, right?

Reply:
No, no, I'm replying to Andreas; `build-root-filesystem.sh` was doing everything correctly, and so does this code.

Reply:
Oh yeah lol, my mistake :smile: